File: app/src/main/java/ntktest/example/com/ntktest/data/RssItem.java
package ntktest.example.com.ntktest.data;

import android.support.annotation.NonNull;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Root(strict = false, name = "item")
public class RssItem {
    @NonNull
    @Element(required = false, data = true)
    private String title = "";

    @NonNull
    @Element(required = false, data = true)
    private String description = "";

    @NonNull
    @Element(required = false)
    private String link = "";

    @NonNull
    @Element(required = false)
    private String pubDate = "";

    @NonNull
    public String getTitle() {
        return title;
    }

    @NonNull
    public String getLink() {
        return link;
    }

    @NonNull
    public String getDescription() {
        return description;
    }

    @NonNull
    public String getPubDate() {
        return pubDate;
    }

    @NonNull
    public String toString() {
        return "RssItem { "
                + "title=" + title + ", "
                + "link=" + link + " }";
    }
}

File: app/src/main/java/com/example/rssreader/data/RssItem.kt
package com.example.rssreader.data

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

// todo figure out that 'required' thing (look at RssDocument)
@Root(strict = false, name = "item")
data class RssItem(
    @param:Element(name = "title", required = false, data = true)
    @field:Element(name = "title", required = false, data = true)
    val title: String = "",

    @param:Element(name = "link", required = false)
    @field:Element(name = "link", required = false)
    val link: String = "",

    @param:Element(name = "pubDate", required = false)
    @field:Element(name = "pubDate", required = false)
    val pubDate: String = ""
)
